src/Features/Core/Portable/Workspace/CompileTimeSolutionProvider.cs

```diff
@@ -67,12 +66,15 @@ public CompileTimeSolutionProvider(Workspace workspace)
                 {
                     lock (_gate)
                     {
-                        _primaryBranchCompileTimeCache = null;
-                        _forkedBranchCompileTimeCache = null;
+#if NETCOREAPP
+                        _designTimeToCompileTimeSolution.Clear();
+#else
+                        _designTimeToCompileTimeSolution = new();
```

Comment:
note: doing the above seems unnecessary.  as a CWT, they should naturally release their resources when they are no longer referenced anywhere.